src/main/java/net/mcreator/generator/template/TemplateExpressionParser.java

```diff
@@ -123,17 +97,41 @@ private static boolean parseCondition(@Nonnull Generator generator, @Nonnull Str
 		return false;
 	}
 
-	private static boolean processFTLExpression(Generator generator, String expression) {
+	private static Object getValueFrom(String field, Object conditionDataProvider) throws ReflectiveOperationException {
+		if (!field.contains("()")) { // field
+			return conditionDataProvider.getClass().getField(field.trim()).get(conditionDataProvider);
+		} else { // method
+			return conditionDataProvider.getClass().getMethod(field.replace("()", "").trim())
+					.invoke(conditionDataProvider);
+		}
+	}
+
+	public static Object processFTLExpression(Generator generator, String expression, Object dataHolder) {
 		try {
-			Template t = new Template("INLINE EXPRESSION", new StringReader(expression),
+			Map<String, Object> dataModel = new HashMap<>(generator.getBaseDataModelProvider().provide());
+			AtomicReference<?> retVal = new AtomicReference<>(null);
+			dataModel.put("retVal", retVal);

```

Comment:
Change retVal to _retVal so _ indicates variable for internal use, to prevent collision if user uses retVal in expression

Reply:
:heavy_check_mark: Fixed